usvm-python/usvm-python-main/src/main/kotlin/org/usvm/machine/interpreters/symbolic/operations/basic/MethodNotifications.kt

```diff
@@ -37,7 +38,10 @@ fun nbAddKt(
     context.ctx
 ) {
     context.curState ?: return
-    pyAssert(context, left.evalIsSoft(context, HasNbAdd) or right.evalIsSoft(context, HasNbAdd))
+    val nbAdd = left.evalIsSoft(context, HasNbAdd) or right.evalIsSoft(context, HasNbAdd)
+    val sqConcat = left.evalIsSoft(context, HasSqConcat) and right.evalIsSoft(context, HasSqConcat)
+    pyAssert(context, context.ctx.mkImplies(nbAdd.not(), sqConcat))
```

Comment:
`context.ctx.mkImplies(nbAdd.not(), sqConcat)` -> `nbAdd.not() implies sqConcat`

Reply:
Please add comment that describes the logic behind these changes.

Reply:
Also, please add the method that catches the event of `sqConcat` (similar to `nbAdd`). In that method, add constraint that operands have `sq_concat` method. In some cases, execution might skip `nbAdd` and go straight to `sqConcat`, but we still need to fix that operands have this method.